### PR TITLE
Grid improvements

### PR DESCRIPTION
**Pull request by conor-cafferkey-sociomantic**
A few updates to Grid and GridItem:
- Grid:
    - Grid now supports `(align|justify)Content` as well as `(align|justify)Items`
    - Renamed some Grid props to match the naming of corresponding CSS Grid properties:
        - `cols` ➡️ `columns`
    - Sets most style properties inline and gets spacings from `useThemeVars()`
    - ~Removed `custom(Rows|Columns)` props; row/column template can now be set by passing an array to `rows` or `columns` props (_does this makes sense?_)~ Renamed `custom(Rows|Columns)` to `template(Rows|Columns)`
    - Removed `(row|column)Gap` props; added new `gap` prop that sets both (accepts a single value or a tuple)
    - Updated docs
- GridItem:
    - Removed wrapper div if `children` is either a function (render prop) or a single React element and inject `style` prop directly into child element
        - still wraps in a div if `children` is _not_ a function or element
    - Updated docs

__❗️All of these changes are open for discussion, _please_ let me know if they make sense❗️__

### CODE REVIEW

File: src/Grid/index.jsx [hunk 1]
```diff
@@ -51,85 +66,92 @@ const Grid = forwardRef( ( props, ref ) =>
...
-    cssMap        : PropTypes.objectOf( PropTypes.string ),
+    cssMap : PropTypes.objectOf( PropTypes.string ),
     /**
-     *  Custom sizes of columns
+     *  Gap between rows and/or columns
      */
-    customColumns : PropTypes.string,
+    gap    : PropTypes.oneOfType(
+        PropTypes.oneOf( [ 'none', 's', 'm', 'l' ] ),
+        PropTypes.arrayOf( [ PropTypes.oneOf( [ 'none', 's', 'm', 'l' ] ) ] ),
```
Comment (conor-cafferkey-sociomantic) [inline src/Grid/index.jsx]: Changed these to lowercase to match what we export in theme vars... problem is this is inconsistent with what we have else where (we’ve generally used uppercase). I guess we should either:
- update the theme vars export to be uppercase, or
- update all component props to use lowercase

... thoughts?

Comment (daniel-martic-sociomantic) [reply src/Grid/index.jsx]: I think latter is better: update all component props to use lowercase - most components already have lowercase prop values whenever prop value is a string or an array of strings.

**Comment by conor-cafferkey-sociomantic** [comment]
> Removed `custom(Rows|Columns)` props; row/column template can now be set by passing an array to `rows` or `columns` props (_does this makes sense?_)

I think this probably doesn’t make sense after all. After some thought I reckon we should just keep it simple:
- `rows` and `columns` props accept only a number (as before)
- rename `custom(Rows|Columns)` to `template(Rows|Columns)` to match the native CSS Grid naming
- no arrays nonsense

@daniel-martic-sociomantic what do you think?

**Comment by daniel-martic-sociomantic** [comment]
I agree - let's keep it simple 👍

**Comment by mahmoud-zakria-sociomantic** [comment]
Booom 💥 👍

**Pull request merged**